From 4b86f141c7c4725093fb2e12c769fec393727519 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@gmail.com>
Date: Sun, 28 Aug 2016 21:04:01 +0200
Subject: [PATCH] compat: set NETDB_INTERNAL

Musl doesn't provide NETDB_INTERNAL which is defined in resolv/netdb.h
in Glibc [1].

Set NETDB_INTERNAL to -1 locally if not already defined.
Based on [2].

[1] https://sourceware.org/git/?p=glibc.git;a=blob;f=resolv/netdb.h;h=3aba530932c7a62a4f23e3193e9186da677f552b;hb=fdfc9260b61d3d72541f18104d24c7bcb0ce5ca2#l74
[2] http://git.alpinelinux.org/cgit/aports/tree/main/socat/netdb-internal.patch?id=5a45173b50892cb634197c30b3506ebff98d3b7d

Upstream: sent to socat@dest-unreach.org
Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 compat.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compat.h b/compat.h
index bb7e2d8f1782..72e73c701f64 100644
--- a/compat.h
+++ b/compat.h
@@ -851,6 +851,8 @@ typedef unsigned long T_sigset;
 
 #if !defined(NETDB_INTERNAL) && defined(h_NETDB_INTERNAL)
 #  define NETDB_INTERNAL h_NETDB_INTERNAL
+#elif !defined(NETDB_INTERNAL)
+#  define NETDB_INTERNAL (-1)
 #endif
 
 #ifndef INET_ADDRSTRLEN
-- 
2.5.5

